### PR TITLE
Fix staking UI issues (from #2292)

### PR DESCRIPTION
**Pull request by liberumed**
## Motivation
This PR is made to fix remin WEB UI issues for POSDAO Staking DApp, directly https://github.com/poanetwork/blockscout/pull/2292#issuecomment-522098750
This is a more clean version of https://github.com/poanetwork/blockscout/pull/2618 , so should contain commits for each fix and remove some unnecessary logic. 
  
## Changelog
- [x] Fix items disappearing misbehavior of staking filters (5, 26)
- [x] Fix items disappearing animation of staking filters (7)
- [x] Fix not disappearing tooltip in stakes-top container on block number update (3)
- [x] Fix stakes top stats width to have fixed size on refresh (4)
- [x] Replace stakes table with flex boxes to fixate size of table header (6, 13)
- [x] Fix staking modals width (2, 12)

### CODE REVIEW

File: apps/block_scout_web/assets/js/pages/stakes.js [hunk 1]
```diff
@@ -101,6 +102,7 @@ if ($stakesPage.length) {
   store.dispatch({ type: 'CHANNEL_CONNECTED', channel })
 
   channel.on('staking_update', msg => {
+    $('.tooltip').hide()
```
Comment (goodsoft) [inline apps/block_scout_web/assets/js/pages/stakes.js]: I would rather limit the search by the top panel:
```suggestion
    $stakesTop.find('.tooltip').hide()
```

And separately handle tooltips in the pools table (probably, globally in the `async_listing_load.js`)

File: apps/block_scout_web/assets/css/components/stakes/_stakes.scss [hunk 2]
```diff
@@ -119,7 +130,17 @@ $stakes-stats-item-border-color: #fff !default;
   }
 }
 
+.stakes-top-stats-item {
+  min-width: 180px;
+
+  &.stakes-top-stats-item-address {
+    min-width: 390px;
```
Comment (goodsoft) [inline apps/block_scout_web/assets/css/components/stakes/_stakes.scss]: This line is useless in Chrome, but pushes the "Become a Candidate" button out-of-screen in Firefox.
```suggestion
```

File: apps/block_scout_web/assets/css/components/stakes/_modal_delegators_info.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+.modal-delegators-info {
+  max-width: map-get($container-max-widths, "xl");
+  min-width: map-get($container-max-widths, "md");
```
Comment (goodsoft) [inline apps/block_scout_web/assets/css/components/stakes/_modal_delegators_info.scss]: The modal window becomes too wide now.

Can we make it only take as much as it needs, or even fixed size?
We even cut delegators' addresses for that purpose.

File: apps/block_scout_web/assets/css/components/stakes/_modal_validator_info.scss [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+  max-width: 660px;
+  width: 100%;
 
   @include media-breakpoint-down(sm) {
     margin-left: auto;
     margin-right: auto;
   }
+
+  .modal-title {
+    margin-bottom: 15px;
```
Comment (goodsoft) [inline apps/block_scout_web/assets/css/components/stakes/_modal_validator_info.scss]: This margin looks gigantic, and also doesn't work reliably between Chrome and Firefox for some reason (probably, order of imports breaks cascading).

File: apps/block_scout_web/lib/block_scout_web/templates/stakes/_stakes_modal_delegators_list.html.eex [hunk 5]
```diff
@@ -1,62 +1,62 @@
...
-            <% end %>
-          </tbody>
-        </table>
+            </div>
+          </div>
+          <%= for {delegator, index} <- Enum.with_index(@delegators, 1) do %>
+            <div class="row">
+              <div class="col-1 stakes-td"><div class="stakes-td-order"><%= index %></div></div>
+              <div class="col-4 stakes-td">
+                <div class="stakes-address-container">
```
Comment (goodsoft) [inline apps/block_scout_web/lib/block_scout_web/templates/stakes/_stakes_modal_delegators_list.html.eex]: Also, removing this class might fix problem 19.

File: apps/block_scout_web/assets/css/components/stakes/_modal_delegators_info.scss (hunk 3 above)
Comment (liberumed) [reply apps/block_scout_web/assets/css/components/stakes/_modal_delegators_info.scss]: But according to design: https://app.zeplin.io/project/5a6053176eb2bda2cf9379be/screen/5d089e95ec53c45d4abcef19
width: 1526px

Comment (goodsoft) [reply apps/block_scout_web/assets/css/components/stakes/_modal_delegators_info.scss]: The design is apparently drawn for 2x pixel density without setting proper DPI or something.
You should divide _all_ sizes by 2.

File: apps/block_scout_web/assets/js/pages/stakes.js (hunk 1 above)
Comment (liberumed) [reply apps/block_scout_web/assets/js/pages/stakes.js]: Tooltip container out of stakesTop context and appear just before closed body tag.

Comment (goodsoft) [reply apps/block_scout_web/assets/js/pages/stakes.js]: _okay.jpg_

File: apps/block_scout_web/assets/css/components/stakes/_stakes.scss (hunk 2 above)
Comment (liberumed) [reply apps/block_scout_web/assets/css/components/stakes/_stakes.scss]: Need at least 260px for non-logged-in state.

File: apps/block_scout_web/assets/css/components/stakes/_modal_validator_info.scss (hunk 4 above)
Comment (liberumed) [reply apps/block_scout_web/assets/css/components/stakes/_modal_validator_info.scss]: Look ok in Firefox and Chrome on Mac and also follow provided design:
https://app.zeplin.io/project/5a6053176eb2bda2cf9379be/screen/5c8bca86161735068c52fc27

File: apps/block_scout_web/assets/css/components/stakes/_stakes.scss (hunk 2 above)
Comment (goodsoft) [reply apps/block_scout_web/assets/css/components/stakes/_stakes.scss]: We have ellipsizing there, so no problem if it is too narrow.
In any case, this line has no effect in Chrome.

File: apps/block_scout_web/assets/css/components/stakes/_modal_validator_info.scss (hunk 4 above)
Comment (goodsoft) [reply apps/block_scout_web/assets/css/components/stakes/_modal_validator_info.scss]: The margin in design is 20px, which should be divided by 2 as well, i.e. 10px.

File: apps/block_scout_web/assets/css/components/stakes/_stakes.scss (hunk 2 above)
Comment (liberumed) [reply apps/block_scout_web/assets/css/components/stakes/_stakes.scss]: ellipsizing not working for me but min-width: 260px; works perfectly in chrome
Screen Shot 2019-09-13 at 20 48 44

**Pull request merged**